lib/ios/AirGoogleMaps/AIRGoogleMap.m

```diff
@@ -528,6 +545,203 @@ + (GMSCameraPosition*) makeGMSCameraPositionFromMap:(GMSMapView *)map andMKCoord
+
+- (AIRGoogleMapMarker*) markerAtPoint:(CGPoint)point {
+    AIRGoogleMapMarker* mrk = nil;
+    for (AIRGoogleMapMarker* mrkView in self.markers) {
+        CGRect frame = [self frameForMarker:mrkView];
+        if (CGRectContainsPoint(frame, point)) {
+            mrk = mrkView;
+            break;
+        }
+    }
+    return mrk;
+}
+
+-(SEL)getActionForTarget:(NSObject*)target {
+    SEL action = nil;
+    uint32_t ivarCount;
+    Ivar *ivars = class_copyIvarList([target class], &ivarCount);
+    if (ivars) {
+        for (uint32_t i = 0 ; i < ivarCount ; i++) {
+            Ivar ivar = ivars[i];
+            const char* type = ivar_getTypeEncoding(ivar);
+            const char* ivarName = ivar_getName(ivar);
+            NSString* name = [NSString stringWithCString: ivarName encoding: NSASCIIStringEncoding];
+            if (type[0] == ':' && [name isEqualToString:@"_action"]) {
+                SEL sel = ((SEL (*)(id, Ivar))object_getIvar)(target, ivar);
+                action = sel;
+                break;
+            }
+        }
+    }
+    return action;
+}
+
+#pragma mark - Overrides for Callout behavior
+
+-(void)overrideGestureRecognizersForView:(UIView*)view {
+    NSArray* grs = view.gestureRecognizers;
+    for (UIGestureRecognizer* gestureRecognizer in grs) {
+        NSNumber* grHash = [NSNumber numberWithUnsignedInteger:gestureRecognizer.hash];
+        if([_origGestureRecognizersMeta objectForKey:grHash] != nil)
+            continue; //already patched
+        
+        //get orig targets
+        NSArray* origTargets = [gestureRecognizer valueForKey:@"targets"];
+        NSMutableArray* origTargetsActions = [[NSMutableArray alloc] init];
+        BOOL isZoomTapGesture = NO;
+        for (NSObject* trg in origTargets) {
+            NSObject* target = [trg valueForKey:@"target"];
+            SEL action = [self getActionForTarget:trg];
+            isZoomTapGesture = [NSStringFromSelector(action) isEqualToString:@"handleZoomTapGesture:"];
+            [origTargetsActions addObject:@{@"target": target, @"action": NSStringFromSelector(action)}];
+        }
+        if (isZoomTapGesture && self.zoomTapEnabled == NO) {
+            [view removeGestureRecognizer:gestureRecognizer];
+            continue;
+        }
+        
+        //replace with ours
+        for (NSDictionary* origTargetAction in origTargetsActions) {
+            NSObject* target = [origTargetAction objectForKey:@"target"];
+            NSString* actionString = [origTargetAction objectForKey:@"action"];
+            SEL action = NSSelectorFromString(actionString);
+            [gestureRecognizer removeTarget:target action:action];
+        }
+        [gestureRecognizer addTarget:self action:@selector(ourMapGestureHandler:)];
+        
+        [_origGestureRecognizersMeta setObject:@{@"targets": origTargetsActions}
+                                        forKey:grHash];
+    }
+}
+
+- (id)ourMapGestureHandler:(UIGestureRecognizer*)gestureRecognizer {
```

Comment:
Maybe find a better name for this method, unless it follows something already in?

Reply:
Renamed to extendedMapGestureHandler